orangecontrib/geo/widgets/owgeotransform.py

```diff
@@ -0,0 +1,153 @@
+                                      contentsLength=100,
+                                      searchable=True,
+                                      model=self.variable_model,
+                                      orientation=Qt.Horizontal)
+
+        box = gui.vBox(self.controlArea, box="Transformation:")
+
+        self.from_combo = gui.comboBox(
+            box, self, "from_idx",
+            contentsLength=100,
+            searchable=True,
+            items=list(self.EPSG_CODES.keys()),
+            label="From:", orientation=Qt.Horizontal,
+            sendSelectedValue=True
+        )
+
+        self.to_combo = gui.comboBox(
+            box, self, "to_idx",
+            contentsLength=100,
+            searchable=True,
+            items=list(self.EPSG_CODES.keys()),
+            label="To:", orientation=Qt.Horizontal,
+            sendSelectedValue=True
+        )
+
+        self.commit_button = gui.button(self.controlArea, self, "&Commit",
+                                        self.apply)
+
+    def init_attr_values(self):
+        self.Error.clear()
+        lat, lon = None, None
+        if self.data is not None:
+            lat, lon = find_lat_lon(self.data, filter_hidden=True)
+            if lat is None or lon is None:
+                # we either find both or we don't have valid data
+                self.Error.no_lat_lon_vars()
+                self.data = None
+                lat, lon = None, None
+
+        self.variable_model.set_domain(self.data.domain if self.data else None)
+        self.attr_lat, self.attr_lon = lat, lon
+
+    @Inputs.data
+    def set_data(self, data: Table = None):
+        self.data = data
+        self.closeContext()
+
+        if data is None or not len(data):
+            self.clear()
+            self.Outputs.data.send(None)
+            return
+
+        self.init_attr_values()
+        self.transform_to = self.EPSG_CODES[self.from_idx]
+        self.transform_to = self.EPSG_CODES[self.to_idx]
+        self.openContext(self.data)
+        self.apply()
+
+    def clear(self):
+        self.data = None
+        self.attr_lat = self.attr_lon = None
+        self.transform_from, self.transform_to = None, None
+
+    def apply(self):
+        if self.data is None or not len(self.data):
+            self.Outputs.data.send(None)
+            return
+        self.transform_from = self.EPSG_CODES[self.from_idx]
+        self.transform_to = self.EPSG_CODES[self.to_idx]
+        transformer = Transformer.from_crs(self.transform_from,
+                                           self.transform_to)
+        latitude = self.data.get_column_view(self.attr_lat)[0]
+        longitude = self.data.get_column_view(self.attr_lon)[0]
+        lat, lon = zip(*transformer.itransform(zip(latitude, longitude)))
+
+        out = self.data.add_column(ContinuousVariable("lat"), lat)
+        out = out.add_column(ContinuousVariable("lon"), lon)
+        self.Outputs.data.send(out)
```

Comment:
Some (me, without overthinking) may expect the widget to make the transformation and then replace the original Lat/Lon values with the new values. Similar to what the preprocessors do.

But that is only my opinion. If you find your way more intuitive, feel free to keep it.

Reply:
Replace as in create a copy of the table, remove old lat/lon and add new ones? Sure, I can do that.